honeybee_energy/simulation/sizing.py

```diff
@@ -7,30 +7,59 @@
 
 from honeybee.typing import float_positive
 
+from ladybug.ddy import DDY
+from ladybug.designday import DesignDay
+from ladybug.location import Location
+
 
 class SizingParameter(object):
     """Global sizing parameters dictating scaling factors for all zone peak loads.
 
     Properties:
+        * design_days
         * heating_factor
         * cooling_factor
     """
-    __slots__ = ('_heating_factor', '_cooling_factor')
+    __slots__ = ('_design_days', '_heating_factor', '_cooling_factor')
 
-    def __init__(self, heating_factor=1.25, cooling_factor=1.15):
+    def __init__(self, design_days=None, heating_factor=1.25, cooling_factor=1.15):
         """Initialize SizingParameter.
 
         Args:
+            design_days: An array of Ladybug DesignDay objects that represent the
+                criteria for which the HVAC systems will be sized. Default: None.
             heating_factor: A number that will get multiplied by the peak heating load
                 for each zone in the model in order to size the heating system for
                 the model. Must be greater than 0. Default: 1.25.
             cooling_factor: A number that will get multiplied by the peak cooling load
                 for each zone in the model in order to size the cooling system for
                 the model. Must be greater than 0. Default: 1.15.
         """
+        self.design_days = design_days
         self.heating_factor = heating_factor
         self.cooling_factor = cooling_factor
 
+    @property
+    def design_days(self):
+        """Get or set an array of Ladybug DesignDay objects for sizing criteria."""
+        return tuple(self._design_days)
+
+    @design_days.setter
+    def design_days(self, value):
```

Comment:
I personally find having the combination of setter and `add_*` methods confusing. I think you can just have a range of `add_*` methods that one can use to add design days after initiating the object.

Also it might makes sense to have methods to remove_design_days or clear_design_days?

Reply:
I'll add methods to remove design days.

The code of the setter still needs to be there, whether it's in `__init__` or it's in this setter here. So, whether it's public or private really isn't changing much in my view.

Do you not agree with the philosophy of "whatever can be public should be public"?  Personally, I have just found myself in many more situations wishing something was exposed rather than wishing it was hidden.

Reply:
My only concern is that a setter can overwrite all the other values in the list in this case. Generally speaking I agree with having what can be public, public.

Reply:
I think that's just the nature of working with a powerful API that you can overwrite things. As they say, "with great power comes great responsibility."